### PR TITLE
New scanner firmware

### PR DESCRIPTION
**Pull request by hslusarek**
This PR adds support for the new scanner firmware.

### CODE REVIEW

File: include/psen_scan/start_request.h [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+/**
+ * @brief Frame containing all necessary fields for a Start Request.
+ *
+ * Unless otherwise indicated the byte order is little endian.
+ *
+ */
+typedef struct StartRequest
+{
+public:
```
Comment (hslusarek) [inline include/psen_scan/start_request.h:36]: Please, make these variables private (if possible).

File: include/psen_scan/start_request.h [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  uint16_t slave2_start_angle_;
+  uint16_t slave2_end_angle_;
+  uint16_t slave2_resolution_;
+  uint16_t slave3_start_angle_;
+  uint16_t slave3_end_angle_;
+  uint16_t slave3_resolution_;
+
+public:
+  StartRequest(const ScannerConfiguration& scanner_configuration)
+  : target_udp_port_(htobe16(scanner_configuration.target_udp_port_))
```
Comment (hslusarek) [inline include/psen_scan/start_request.h]: - Please, access the scanner configuration via getter functions.

File: include/psen_scan/start_request.h [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  uint8_t io_pin_enabled_;
+  uint8_t scan_counter_enabled_;
+  uint8_t speed_encoder_enabled_;    /**< 0000000bin disabled, 00001111bin enabled.*/
+  uint8_t diagnostics_enabled_;
+
+  /**< The following 'angle' and 'resolution' fields are all in tenth of degrees */
+  uint16_t master_start_angle_;
+  uint16_t master_end_angle_;
+  uint16_t master_resolution_;
+  uint16_t slave1_start_angle_;
```
Comment (hslusarek) [inline include/psen_scan/start_request.h]: I suggest using one array here for the different `slaveK_start_angle_`, `slaveK_end_angle_` and `slaveK_resolution_` variables. This way it is easier to add slaves. A getter function could then simplify the access to the different angle variables of the slaves.

File: include/psen_scan/start_request.h [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  , slave2_resolution_(htole16(scanner_configuration.slave2_resolution_))
+  , slave3_start_angle_(htole16(scanner_configuration.slave3_start_angle_))
+  , slave3_end_angle_(htole16(scanner_configuration.slave3_end_angle_))
+  , slave3_resolution_(htole16(scanner_configuration.slave3_resolution_))
+  {
+    // TODO: What to do with seq_number???
+    setTargetIP(scanner_configuration.target_ip_);
+    setEnableFields(scanner_configuration);
+    setCRC();
+  }
```
Comment (hslusarek) [inline include/psen_scan/start_request.h]: For easier readability of the interface, I suggest to split declaration and definition of the functions.

File: include/psen_scan/start_request.h [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+{
+
+/**
+ * @brief Frame containing all necessary fields for a Start Request.
+ *
+ * Unless otherwise indicated the byte order is little endian.
+ *
+ */
+typedef struct StartRequest
+{
```
Comment (hslusarek) [inline include/psen_scan/start_request.h:35]: For an easier readability of the interface I suggest to first introduce the functions and then the variables.

File: include/psen_scan/start_request.h [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+namespace psen_scan
+{
+
+/**
+ * @brief Frame containing all necessary fields for a Start Request.
+ *
+ * Unless otherwise indicated the byte order is little endian.
+ *
+ */
+typedef struct StartRequest
```
Comment (hslusarek) [inline include/psen_scan/start_request.h]: I suggest to use a class. Why?
- Our default intention is to hide things from the higher abstractions level which contradicts to some extend the default intention of a struct.
- See also: 
  - https://www.fluentcpp.com/2017/06/13/the-real-difference-between-struct-class/
  - https://www.geeksforgeeks.org/structure-vs-class-in-cpp/
  - http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-org

What do you think?

File: include/psen_scan/crc.h [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+#include <boost/crc.hpp>
+
+namespace psen_scan
+{
+
+namespace crc
+{
+
+  inline uint32_t calcCRC32(const void* data_buffer, const size_t& byte_count)
```
Comment (hslusarek) [inline include/psen_scan/crc.h]: Static or inline? What's is better?

File: include/psen_scan/start_request.h (hunk 1 above)
Comment (SansoneG) [reply include/psen_scan/start_request.h:36]: done

File: include/psen_scan/start_request.h (hunk 4 above)
Comment (SansoneG) [reply include/psen_scan/start_request.h]: done

File: include/psen_scan/start_request.h (hunk 5 above)
Comment (SansoneG) [reply include/psen_scan/start_request.h:35]: done

File: include/psen_scan/start_request.h (hunk 3 above)
Comment (SansoneG) [reply include/psen_scan/start_request.h]: done

File: include/psen_scan/start_request.h (hunk 2 above)
Comment (SansoneG) [reply include/psen_scan/start_request.h]: done

**Comment by hslusarek** [comment]
Development of the new firmware has moved to new repository: https://github.com/PilzDE/psen_scan_v2.
-> PR is closed.

**Pull request closed without merging**